File: My_projects/DNA_translator/DNA_Translator_Java/src/main/java/DNA_Translator.java
import org.biojava.nbio.core.sequence.DNASequence;
import org.biojava.nbio.core.sequence.io.FastaReaderHelper;
import org.biojava.nbio.core.sequence.template.SequenceView;
import org.biojava.nbio.core.sequence.transcription.TranscriptionEngine;
import org.biojava.nbio.core.sequence.ProteinSequence;
import org.biojava.nbio.core.sequence.RNASequence;

import java.io.File;
import java.util.LinkedHashMap;

public class DNA_Translator {
    public static void main(String[] args) {
        long startTime = System.nanoTime();

        try {
            // Load DNA sequence from FASTA file
            File fastaFile = new File("/Users/giuse/pythonProject/Mycodes/samples/sequence1.txt");
            LinkedHashMap<String, DNASequence> sequences = FastaReaderHelper.readFastaDNASequence(fastaFile);

            // Get the sequence
            DNASequence dna = sequences.values().iterator().next();
            System.out.println("Loaded DNA Sequence: " + dna.getSequenceAsString());

            // Get Reverse Complement
            SequenceView reverseComplementView = dna.getReverseComplement();
            DNASequence reverseComplement = new DNASequence(reverseComplementView.getSequenceAsString());
            System.out.println("Original DNA Sequence: " + dna.getSequenceAsString());
            System.out.println("Reverse Complement: " + reverseComplement.getSequenceAsString());


            // Transcribe DNA to RNA
            TranscriptionEngine engine = TranscriptionEngine.getDefault();
            RNASequence rna = dna.getRNASequence(engine);
            System.out.println("Transcribed mRNA: " + rna.getSequenceAsString());

            // Translate RNA to Protein
            ProteinSequence protein = rna.getProteinSequence(engine);
            System.out.println("Protein Sequence: " + protein.getSequenceAsString());

        } catch (Exception e) {
            e.printStackTrace();
        }
        long endTime = System.nanoTime();
        long duration = endTime - startTime;
        System.out.println("Execution Time " + duration / 1_000_000 + "ms");
    }
}

File: My_projects/DNA_translator/DNA_translator_R/DNA_translator.R
# DNA Sequence Translator
library(Biostrings)

start_time <- Sys.time()
# Load FASTA
fasta_file <- "/Users/giuse/pythonProject/Mycodes/samples/sequence1.txt"
dna_sequence <- readDNAStringSet(fasta_file)

# Print Original DNA Sequence
cat("Original DNA sequence:\n")
print(dna_sequence)

# Get Reverse Complement
reverse_complement <- reverseComplement(dna_sequence)
cat("\nReverse Complement:\n")
print(reverse_complement)

# Transcribe to mRNA (DNA -> RNA)
rna_sequence <- RNAStringSet(dna_sequence)
cat("\nTranscribed mRNA:\n")
print(rna_sequence)

# Translate (RNA -> Protein)
protein_sequence <- translate(dna_sequence)
cat("\nProtein Sequence:\n")
print(protein_sequence)

end_time <- Sys.time()
exec_time <- end_time - start_time
print(exec_time * 1000) # milliseconds